File: Dread's Projects/DreadGame/GameData/GameData.cpp
#include "GameData.h"

GameData::GameData()
{
    player = nullptr;
    NPC = new NPC();
}

void GameData::CreatePlayer(string name)
{
    player = new Player(name);
}

const string GameData::getPlayerName()
{
    return player->getName();
}

int GameData::getPlayerHP()
{
    return player->getHP();
}

int GameData::getPlayerDamage()
{
    return player->getDamage();
}

int GameData::getMonsterHP()
{
    return monster->getHP();
}

int GameData::getMonsterDamage()
{
    return monster->getDamage();
}

GameData::~GameData() {}

File: Java/Monster Slayer Java Edition/src/GameLogic/GameData.java
package GameLogic;

public class GameData
{
    private Player player;
    private Monster monster;
    
    public GameData()
    {
        this.player = new Player();
        this.monster = new Monster();
    }
    
    public int getPlayerHP() { return this.player.getHP(); }
    public int getMonsterHP() { return this.monster.getHP(); }
    
    public void setPlayerName(String name) { player.SetPlayerName(name); }
    
    public boolean isOver()
    {
        if(getMonsterHP() == 0 || getPlayerHP() == 0)
            return true;
        else
            return false;
    }
}